### PR TITLE
Remove the file extensions from the bundledStyleNames method

### LINKED ISSUE #969: bundledStyleNames inconsistency
In MGLMapView.mm, the bundledStyleNames method returns an array of filename with the .json extension included. However, the useBundledStyleNamed:(NSString *)styleName expects a filename without the extension.

This makes getting a list of names, selecting one, and call useBundledStyleNamed: more difficult than it needs to be.

I would be happy to make the change, if provided with direction on either including or excluding the filename extension.

### PR DESCRIPTION
**Pull request by picciano**
fixes #969

### CODE REVIEW

File: platform/ios/MGLMapView.mm [hunk 1]
```diff
@@ -944,7 +944,13 @@ - (NSArray *)bundledStyleNames
     if (!_bundledStyleNames) {
         NSString *stylesPath = [[MGLMapView resourceBundlePath] stringByAppendingString:@"/styles"];
 
-        _bundledStyleNames = [[[NSFileManager defaultManager] contentsOfDirectoryAtPath:stylesPath error:nil] mutableCopy];
+        _bundledStyleNames = [NSMutableArray array];
+        
+        NSArray *bundledStyleNamesWithExtensions = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:stylesPath error:nil];
+        // Remove file extension for issue 968
+        for (NSString *styleName in bundledStyleNamesWithExtensions) {
+            [_bundledStyleNames addObject:[styleName stringByDeletingPathExtension]];
```
Comment (1ec5) [inline platform/ios/MGLMapView.mm]: Can you combine this with the `for` loop below that synthesizes hybrid styles?

**Comment by 1ec5** [comment]
Otherwise, the change looks good, but please squash this PR down into a single, clean commit. Thanks!

**Comment by picciano** [comment]
I update the PR to include combining the for loops, but was somewhat unsuccessful at squashing all of the commits. I managed to get the last two commits combined, but was getting errors trying to squash the others. Please advise.

**Comment by kkaefer** [comment]
@picciano please fetch master from this repository, then rebase your branch on that master branch, and force push into your branch. For the future, I recommend not making pull requests from your master branch, but from a feature branch in your repository.

**Comment by kkaefer** [comment]
Even better: do the above, then do a squash rebase: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by picciano** [comment]
rebase was being "difficult", I closed this PR and made a clean one, see #985

**Comment by 1ec5** [comment]
@picciano, in the future, it may help to make the changes on a dedicated branch other than `master` to avoid noise.

**Comment by picciano** [comment]
Agree and noted.

**Pull request closed without merging**